federation-js/src/service/__tests__/printCoreSchema.test.ts

```diff
@@ -229,70 +252,70 @@ describe('printCoreSchema', () => {
-        goodAddress: Boolean @resolve(graph: \\"reviews\\") @requires(fields: \\"{ metadata { address } }\\")
+        id: ID! @join__field(graph: ACCOUNTS)
+        name: Name @join__field(graph: ACCOUNTS)
+        username: String @join__field(graph: ACCOUNTS)
+        birthDate(locale: String): String @join__field(graph: ACCOUNTS)
+        account: AccountType @join__field(graph: ACCOUNTS)
+        metadata: [UserMetadata] @join__field(graph: ACCOUNTS)
+        goodDescription: Boolean @join__field(graph: INVENTORY, requires: \\"metadata { description }\\")
+        vehicle: Vehicle @join__field(graph: PRODUCT)
+        thing: Thing @join__field(graph: PRODUCT)
+        reviews: [Review] @join__field(graph: REVIEWS)
+        numberOfReviews: Int! @join__field(graph: REVIEWS)
+        goodAddress: Boolean @join__field(graph: REVIEWS, requires: \\"metadata { address }\\")
       }
 
       type UserMetadata {
-        name: String
-        address: String
-        description: String
+        name: String @join__field(graph: ACCOUNTS)
+        address: String @join__field(graph: ACCOUNTS)
+        description: String @join__field(graph: ACCOUNTS)
       }
 
       type Van implements Vehicle
-        @owner(graph: \\"product\\")
-        @key(fields: \\"{ id }\\", graph: \\"product\\")
-        @key(fields: \\"{ id }\\", graph: \\"reviews\\")
+        @join__owner(graph: PRODUCT)
+        @join__type(graph: PRODUCT, key: \\"id\\")
+        @join__type(graph: REVIEWS, key: \\"id\\")
       {
-        id: String!
-        description: String
-        price: String
-        retailPrice: String @resolve(graph: \\"reviews\\") @requires(fields: \\"{ price }\\")
+        id: String! @join__field(graph: PRODUCT)
+        description: String @join__field(graph: PRODUCT)
+        price: String @join__field(graph: PRODUCT)
+        retailPrice: String @join__field(graph: REVIEWS, requires: \\"price\\")
       }
 
       interface Vehicle {
-        id: String!
-        description: String
-        price: String
-        retailPrice: String
+        id: String! @join__field(graph: PRODUCT)
+        description: String @join__field(graph: PRODUCT)
+        price: String @join__field(graph: PRODUCT)
+        retailPrice: String @join__field(graph: PRODUCT)
       }
       "
     `);
   });
 
-  it('fieldsets are parseable', () => {
-    const parsedCsdl = parse(coreSchema!);
-    const fieldSets: string[] = [];
-
-    // Collect all args with the 'fields' name (from @key, @provides, @requires directives)
-    visit(parsedCsdl, {
-      Argument(node) {
-        if (node.name.value === 'fields') {
-          fieldSets.push((node.value as StringValueNode).value);
-        }
-      },
-    });
-
-    // Ensure each found 'fields' arg is graphql parseable
-    fieldSets.forEach((unparsed) => {
-      expect(() => parse(unparsed)).not.toThrow();
-    });
-  });
+  // xit('fieldsets are parseable', () => {
```

Comment:
Do we need a replacement for this, or another way of testing output for field sets?

Reply:
Yep! Reintroduced: 2313d5a